java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -94,10 +95,41 @@ public JObject(
       for (Map.Entry<String, JSONSchemaProps> field : fields.entrySet()) {
         String nextPrefix = (config.getPrefixStrategy() == Config.Prefix.ALWAYS) ? classPrefix : "";
         String nextSuffix = (config.getSuffixStrategy() == Config.Suffix.ALWAYS) ? classSuffix : "";
+        String fieldKey = field.getKey();
+        // let's handle duplicates
+        final Map<String, Map<String, JSONSchemaProps>> duplicates = new HashMap<>();
+        // counting the current field duplicates
+        final List<Map.Entry<String, JSONSchemaProps>> fieldDuplicates = fields.entrySet().stream()
+            .filter(f -> sanitizeString(field.getKey()).equals(sanitizeString(f.getKey())))
+            .collect(Collectors.toList());
+        final Long duplicatesCount = fieldDuplicates.stream().count();
+        final Boolean duplicatesExist = duplicatesCount > 1;
+        if (duplicatesExist) {
+          // we want to throw an exception on some duplicates missing requirements. At the moment, the only one we
+          // enforce is for a (duplicatesCount - 1) number of duplicates to be marked as deprecated
+          final Long deprecatedDuplicates = fieldDuplicates.stream()
+              .filter(d -> d.getValue().getDescription().startsWith(DEPRECATED_FIELD_MARKER)).count();
+          if (deprecatedDuplicates != (duplicatesCount - 1)) {
```

Comment:
I'm perfectly fine with this approach, but I would love to quickly discuss this, have you considered alternative approaches?
E.g. append a count number on duplicates instead of failing

Reply:
The approach I considered allows for more than one deprecated duplicated field, allowing for exactly one non-deprecated duplicated field.
But if you think this use case is not reasonable, we can definitely switch to a simpler check. I don't have a strong opinion, feel free to let me know what you prefer.

Reply:
Personally, I'm fine with this but I will probably prefer allowing non-deprecated duplicates as well.
I don't have a super strong opinion here either, I would love to have more feedback on this cc. @manusa @metacosm @shawkins

Reply:
I eventually decided to push a change that reflects what you initially asked for @andreaTP. 
Having more than one duplicated field seems really to be a rare (and I'd say bad) use case, which shouldn't be supported.
Additionally, this allows for us to enforce a more strict requirement about the reason for having such duplicated fields, i.e. one of them has to be deprecated. 

The implementation looks better this way to me, let me know what you think.

Reply:
I'm ok with starting with the logic proposed here!